dist/bin/current/helperScripts/networkFunctions.sh

```diff
@@ -1,42 +1,59 @@
+    mkdir -p "$runtimeFolder" 2> /dev/null
+  fi
 
-    mkdir -p dist/runtime/ 2>/dev/null
-    (cd ${runtimeFolder} && wget -S $(curl -s https://kobo.github.io/groovyserv/download.html | grep groovyserv | grep bin.zip | sed 's/href=/\n/g' | sed 's/>/\n/g' | sed 's/</\n/g' | sed 's/"//g'  | grep .zip)) &> /dev/null
+  if [[ ! -d $(find ${runtimeFolder} -maxdepth 1 -mindepth 1 -type d -name "groovys*") && ! -f ${forbiddenFile} ]]; then
+    (cd ${runtimeFolder} &&
+        wget -S $(curl -s https://kobo.github.io/groovyserv/download.html \
+        | grep groovyserv \
+        | grep bin.zip \
+        | sed 's/href=/\n/g' \
+        | sed 's/>/\n/g' \
+        | sed 's/</\n/g' \
+        | sed 's/"//g'  \
+        | grep .zip)) &> /dev/null
 
     if [[ $? -ne 0 ]]; then
-      touch ${forbiddenFile}
+        echo "Error downloading groovyserv. Skipping groovyserv setup. Remove ${forbiddenFile} before trying again." > /dev/stderr
+        touch ${forbiddenFile}
     else
-      (cd ${runtimeFolder} && unzip groovyser*.zip && rm groovyser*.zip*)
-      [[ $? -ne 0 ]] && touch ${forbiddenFile}
-      (${runtimeFolder}/groovys*/bin/groovyclient 2&> /dev/null)
-      [[ $? -ne 0 ]] && touch ${forbiddenFile}
+      (cd ${runtimeFolder} && unzip groovyser*.zip && rm groovyser*.zip*) &> /dev/null
+      if [[ $? -ne 0 ]]; then
+        echo "Error unzipping groovyserv. Skipping groovyserv setup. Remove ${forbiddenFile} before trying again." > /dev/stderr
+        touch ${forbiddenFile}
+      fi
+      ${runtimeFolder}/groovys*/bin/groovyclient &> /dev/null
+      if [[ $? -ne 0 ]]; then
+        echo "Error starting groovyserv. Skipping groovyserv setup. Remove ${forbiddenFile} before trying again." > /dev/stderr
+        touch ${forbiddenFile}
+      fi
     fi
   fi
 
-  [[ ! -f ${forbiddenFile} ]] && callerBinary=$(readlink -f $( ls ${runtimeFolder}/groovyserv*/bin/groovyclient 2> /dev/null ) 2> /dev/null) || touch ${forbiddenFile}
+  [[ ! -f ${forbiddenFile} ]] && callerBinary=$(readlink -f $(ls ${runtimeFolder}/groovyserv*/bin/groovyclient) 2> /dev/null) || touch ${forbiddenFile}
   echo $callerBinary
 }
 
 function getExistingOrNewGroovyServPort() {
   # Check first, if GroovyServ is running for the local user!
-  local runningGServInstance=$(ps -ef | grep GroovyServ | grep -v grep | grep `whoami`)
+  local runningGServInstance=$(pgrep -f -a -u $(whoami) GroovyServ)
 
   if [[ -n ${runningGServInstance-} ]]; then
     # Use the existing port, it is passed as an argument to GroovyServ, so we can just get it from that.
 
     # Chaining things together above would be nice, but it fails. If we assign the output above
     # to a variable first, the whitespaces will be reduced to a single character thuse making it
     # easy to use cut and rev
-    echo ${runningGServInstance}  | rev | cut -d " " -f 2 | rev
+    echo ${runningGServInstance} | perl -ne '/.+GroovyServer\.main\(args\)\s--\s(\d+)\s.+/; print "$1\n"'
```

Comment:
Did you try running roddy.sh with few parameters? I got "--useRoddyVersion=current" as value of this. The advantage is simply that this always works (if the port is given as first parameter after the `GroovyServ.main(args) --`.